### PR TITLE
Matter

### PR DESCRIPTION
**Pull request by schoeneberg**
Dear all,

this is the first (preliminary) version of my matterlib implementation. Currently, it's still a bit slow (~10s) due to all the inefficient code surrounding it. I expect to be able to improve the speed quite a bit, depending on how much time I have to work on it. Furthermore, I am not 100% certain yet that the results are correct. They look good, but I had to rewrite quite some parts of the code to make it a stand-alone package (as compared to being intrinsic to CLASS). I will try to continue working on this in my spare time over the next week.

Also, please note that I currently set uses_separability=False, which causes slowdown, and uses_separability=True is faster, but might be more inaccurate in the end, depending on the setup. This will have to be tested further.

Cheers,
Nils Sch.

### CODE REVIEW

File: example.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import numpy as np
```
Comment (damonge) [inline example.py:1]: same here, can you put this file in `scripts_MATTER`?

File: n5k/__init__.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from .calculator_base import N5KCalculatorBase
 from .calculator_ccl import N5KCalculatorCCL
+from .calculator_matter import N5KCalculatorMATTER
```
Comment (damonge) [inline n5k/__init__.py:3]: I think this is repeated (see 4 lines below)

File: n5k/calculator_base.py [hunk 3]
```diff
@@ -48,7 +48,9 @@ def get_tracer_parameters(self):
                         1.607983, 1.689579, 1.772899,
                         1.857700, 1.943754, 2.030887,
                         2.118943])
-        return {'b_g': b_g}
+        A_IA = np.full(5, 0.15)
```
Comment (damonge) [inline n5k/calculator_base.py:51]: do you need this at all? I removed it when merging your branch and everything worked well.

File: utils/python/mattery.pyx [hunk 4]
```diff
@@ -199,6 +199,9 @@ cdef class Matter:
             self.ma.tw_weights[nwd*ntw_ni+itw] = tw_ni_weights[nwd][itw]
           self.ma.tw_max[nwd]=tw_ni[nwd][-1]
           self.ma.tw_min[nwd]=tw_ni[nwd][0]
+        for nwd in range(ntr_ni,ntr_i):
+          self.ma.tw_max[nwd]=tw_i[nwd][-1]
+          self.ma.tw_min[nwd]=tw_i[nwd][0]
```
Comment (damonge) [inline utils/python/mattery.pyx:204]: Is this the main change?

**Review by damonge — CHANGES_REQUESTED** [review]
Just a few quick fixes @schoeneberg . Thanks a lot!

File: utils/python/mattery.pyx (hunk 4 above)
Comment (schoeneberg) [reply utils/python/mattery.pyx:204]: Yep, this is the main change.

File: n5k/calculator_base.py (hunk 3 above)
Comment (schoeneberg) [reply n5k/calculator_base.py:51]: Reverted!

File: n5k/__init__.py (hunk 2 above)
Comment (schoeneberg) [reply n5k/__init__.py]: Reverted!

File: example.py (hunk 1 above)
Comment (schoeneberg) [reply example.py]: Moved!

File: compare_output.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import numpy as np
```
[the remark this answers is not in the document]
Comment (schoeneberg) [reply compare_output.py]: Moved!

File: n5k/calculator_base.py (hunk 3 above)
Comment (damonge) [reply n5k/calculator_base.py:51]: Thanks a lot for the changes @schoeneberg . I don't see this one in the current commit though.

Comment (schoeneberg) [reply n5k/calculator_base.py]: I _hope_ this is fixed now. Let me know if there's any problem after all. In particular, my version of the file now looks like master as far as I am aware

Comment (damonge) [reply n5k/calculator_base.py]: Yep, all good!

**Review by damonge — APPROVED** [review]
LGTM

**Pull request merged**